src/Avalonia.Controls/DefinitionBase.cs

```diff
@@ -730,5 +734,22 @@ static DefinitionBase()
             SharedSizeGroupProperty.Changed.AddClassHandler<DefinitionBase>(OnSharedSizeGroupPropertyChanged);
             PrivateSharedSizeScopeProperty.Changed.AddClassHandler<DefinitionBase>(OnPrivateSharedSizeScopePropertyChanged);
         }
+
+        /// <summary>
+        /// Marks a property on a definition as affecting the parent grid's measurement.
+        /// </summary>
+        /// <param name="properties">The properties.</param>
+        protected static void AffectsParentMeasure(params AvaloniaProperty[] properties)
```

Comment:
I thought this was a standard function on any control... Guess this should be considered in 0.10 for a unified approach on this.

Reply:
there is AffectsParentMeasure already, but it's for controls only, here `DefinitionBase` is not a control so it need custom non control implementation

Reply:
@akunchev good catch!